docker-test-integration.sh

```diff
@@ -19,17 +19,23 @@ function debug {
 
 trap debug ERR
 
-BASE_URL="https://storage.googleapis.com/workflow-cli"
-URL="${BASE_URL}/deis-latest-linux-amd64"
+curl-cli-from-gcs-bucket() {
+	local gcs_bucket="${1}"
+	local base_url="https://storage.googleapis.com/${gcs_bucket}"
+	local url="${base_url}/deis-latest-linux-amd64"
 
-if [[ "${CLI_VERSION}" != "latest" ]]; then
-	URL="${BASE_URL}/${CLI_VERSION}/deis-${CLI_VERSION}-linux-amd64"
-fi
+	if [[ "${CLI_VERSION}" != "latest" ]]; then
+		url="${base_url}/${CLI_VERSION}/deis-${CLI_VERSION}-linux-amd64"
+	fi
+
+	# Download CLI, retry up to 5 times with 10 second delay between each
+	echo "Installing Workflow CLI version '${CLI_VERSION}' via url '${url}'"
+	curl -f --silent --show-error -I "${url}"
```

Comment:
I added the `-f/--fail` flag back in so we could get non-zero exit code from curl if the http response code was >= 400 (see also https://curl.haxx.se/mail/archive-2006-02/0004.html).